tlv/primitive.go

```diff
@@ -362,3 +362,29 @@ func SizeBigSize(val interface{}) SizeFunc {
 		return size
 	}
 }
+
+// EBool encodes a boolean. An error is returned if val is not a boolean.
+func EBool(w io.Writer, val interface{}, buf *[8]byte) error {
```

Comment:
Can we move this primitive type to the types above in the file where all the integers are defined (first the Encoding and followed by the Encoding functions), and also include a EBoolT for custom-like boolean types or is this not really necessary ?

Reply:
Thanks for the review. I noticed that the new functions I added in this file should be above the `SizeBigSize` function, is that what you are referring to? as for `EBoolT`, never thought about that if you feel that there will be a good use case fir that, I will include it.

Reply:
Yes either before `SízeBigSize` but I even recommend to sort it anywhere after the `func DUint64` and before `func EBytes32` because to me a bool is way more equal to the basic type like the integers rather than bytes (arrays) and custom types which follow after them.
Yeah I mean I would add `EBoolT` because if you know you are encoding a boolean you could save a heap allocation as well so better have it and stay inline with the other primitive golang types, but I have not strong feeling about it.